File: week02/db_query/frontend/src/types/query.ts
export interface ColumnInfo {
  name: string;
  type: string;
}

export interface QueryResult {
  columns: ColumnInfo[];
  rows: Record<string, unknown>[];
  rowCount: number;
  truncated: boolean;
}

export interface QueryRequest {
  sql: string;
}

export interface NaturalLanguageRequest {
  question: string;
}

export interface GeneratedQuery {
  sql: string;
  explanation: string;
}

File: week02/db_query/backend/src/models/query.py
from typing import Any

from src.models import BaseResponseModel


class QueryRequest(BaseResponseModel):
    sql: str


class ColumnInfo(BaseResponseModel):
    name: str
    type: str


class QueryResultResponse(BaseResponseModel):
    columns: list[ColumnInfo]
    rows: list[dict[str, Any]]
    row_count: int
    truncated: bool
